x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/HashAggregationOperator.java

```diff
@@ -61,6 +71,19 @@ public String describe() {
 
     private final DriverContext driverContext;
 
+    /**
+     * Nanoseconds this operator has spent hashing grouping keys.
+     */
+    private long hashNanos;
+    /**
+     * Nanoseconds this operator has spent running the aggregations.
+     */
+    private long aggregationNanos;
+    /**
+     * Count of pages this operator has processed.
+     */
+    private int pagesProcessed;
```

Comment:
Make that a thought rather than a nit - not sure myself whether my suggestion'd be the best way to go.

Reply:
I actually just did that for `Driver` because it's fairly complex. OTOH I like how the status objects are immutable and that'd be kind of tricky to do with a status object. Let me have a look.